src/Documents.ts

```diff
@@ -829,6 +746,128 @@ class Documents implements Pointer.Library, Pointer.Validator, ObjectSchema.Reso
 		return persistedDocument.getFragment( fragment.id );
 	}
 
+	private setDataRequest( uri:string, requestOptions:HTTP.Request.Options, asContainer:boolean ):string {
+		if( RDF.URI.Util.isRelative( uri ) && ! this.context ) throw new Errors.IllegalArgumentError( "This Documents instance doesn't support relative URIs." );
+
+		if ( this.context ) {
+			if ( this.context.auth.isAuthenticated() ) this.context.auth.addAuthentication( requestOptions );
+			uri = this.context.resolve( uri );
```

Comment:
It's weird that this method is in charge of both configuring the requestOptions and resolving the uri. Please move the uri logic to another method, something like `this.getRequestURI()`